defi_protocols/Votium.py

```
@@ -0,0 +1,105 @@
+    The uncompleted idea is to check for past round numbers. As we are based on json files this is brittle and needs
+    to be changed.
+
+    Args:
+        token_symbol (str): Symbol of the token. eg LDO, FLX
+        round_number (str, optional): The number of the round. Has to be a str in the format 'xxxx'
+          eg 0001 or 0023  Defaults to 'latest'.
+
+    Returns:
+        tuple: (token amount, Claim status)
+    """
+    wallet = Web3.to_checksum_address(wallet)
+    if round_number == 'latest':
+        round_number = token_symbol
+
+    raw_json_url = f'https://raw.githubusercontent.com/oo-00/Votium/main/merkle/{token_symbol}/{round_number}.json'
+    response = requests.get(raw_json_url)
+
+    if response.status_code == 200:
+        json_data = response.json()
+    else:
+        print(f'Failed to download the JSON file. Status code: {response.status_code}')
+    try:
+        index_number = json_data['claims'][wallet]['index']
+        hexstr = json_data['claims'][wallet]['amount']  # get hex amount
+        token_amount = Web3.to_int(hexstr=hexstr) / pow(10, decimals)
+        if check_claimed_or_unclaimed(wallet, index_number):
+            token_amount = 0
+
+        return token_amount
+    except KeyError as e:
+        print(e)
+        pass
+
+
+def check_claimed_or_unclaimed(wallet, index_number):
+    """Interact with the contract to see if the funds have been already claimed or not.
+
+    Args:
+        wallet (str): Wallet address to check.
+        index_number (int): Index number to get the bool value.
+
+    Returns:
+        bool: if claimed or not.
+    """
+    wallet = Web3.to_checksum_address(wallet)
+    claimed = False
+    try:
+        contract = get_contract(contract_address, 'ethereum')
+        claimed = contract.functions.isClaimed(wallet, index_number).call()
+    except Exception as e:
+        print(f'It was not possible to get the contract: {e}')
+
+    return claimed
```

Let the exception propagate so that the caller can do something with it. Because printing is not a way to communicate with "the user of the code" as it can't read the terminal with code to know what happened.
Also claimed will be returning False, but it is not true that it is False, it is unknown.